js/CobrowseIO.js

```diff
@@ -114,11 +114,22 @@ export default class CobrowseIO {
 // the session.requested event is considered internal, it should
 // not be used outside these bindings
 CobrowseIO.addListener('session.requested', (session) => {
+  if (Platform.OS === 'ios' && CobrowseIO.handleFullDeviceRequest) {
+    CobrowseIONative.overwriteFullControlUI()
+  }
+
   CobrowseIO.handleSessionRequest(session)
 })
 
 CobrowseIO.addListener('session.updated', (session) => {
   if (session.isActive() && session.remote_control === 'requested') {
     CobrowseIO.handleRemoteControlRequest(session)
+  } else if (session.isActive() && session.full_device_state === 'requested') {
+    if (CobrowseIO.handleFullDeviceRequest) {
+      CobrowseIO.handleFullDeviceRequest(session)
+    } else if (Platform.OS === 'android') {
+      // accept the incoming connection by default
+      session.setFullDevice('on')
```

Comment:
I don't think this should be required? What happens without this?

Reply:
I think the native prompt is only displayed when we call this. will double check

Reply:
On Android we always overwrite the native functionality. Waiting for setting the full_device to 'on' is what allows us to show the RN UI. When it's not overwritten we need to manually set this to on

Reply:
I've confirmed that this is indeed needed as per the above. The alternative is to use the same implementation as the new PR to conditionally implement the delegate but this would require merging both PRs and, IMO, it's uneeded complexity for this use case